addons/mrp/tests/test_workorder_operation.py

```diff
@@ -2,18 +2,55 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
 from datetime import datetime, timedelta
-
-from odoo.addons.mrp.tests.common import TestMrpCommon
-from odoo.exceptions import UserError
 from odoo.tests import Form
+from odoo.addons.mrp.tests.common import TestMrpCommon
+from odoo.exceptions import ValidationError, UserError
 
 
 class TestWorkOrderProcess(TestMrpCommon):
+    def full_availability(self):
+        """set full availability for all calendars"""
+        self.env['resource.calendar'].search([]).write({'attendance_ids': [
+            (0, 0, {'name': 'Monday', 'dayofweek': '0', 'hour_from': 0, 'hour_to': 24, 'day_period': 'morning'}),
+            (0, 0, {'name': 'Tuesday', 'dayofweek': '1', 'hour_from': 0, 'hour_to': 24, 'day_period': 'morning'}),
+            (0, 0, {'name': 'Wednesday', 'dayofweek': '2', 'hour_from': 0, 'hour_to': 24, 'day_period': 'morning'}),
+            (0, 0, {'name': 'Thursday', 'dayofweek': '3', 'hour_from': 0, 'hour_to': 24, 'day_period': 'morning'}),
+            (0, 0, {'name': 'Friday', 'dayofweek': '4', 'hour_from': 0, 'hour_to': 24, 'day_period': 'morning'}),
+            (0, 0, {'name': 'Saturday', 'dayofweek': '5', 'hour_from': 0, 'hour_to': 24, 'day_period': 'morning'}),
+            (0, 0, {'name': 'Sunday', 'dayofweek': '6', 'hour_from': 0, 'hour_to': 24, 'day_period': 'morning'}),
+        ]})
 
     def setUp(self):
         super(TestWorkOrderProcess, self).setUp()
         self.source_location_id = self.ref('stock.stock_location_14')
         self.warehouse = self.env.ref('stock.warehouse0')
+        # setting up alternative workcenters
+        self.wc_alt_1 = self.env['mrp.workcenter'].create({
+            'name': 'Nuclear Workcenter bis',
+            'capacity': 3,
+            'time_start': 9,
+            'time_stop': 5,
+            'time_efficiency': 80,
+        })
+        self.wc_alt_2 = self.env['mrp.workcenter'].create({
+            'name': 'Nuclear Workcenter ter',
+            'capacity': 1,
+            'time_start': 10,
+            'time_stop': 5,
+            'time_efficiency': 85,
+        })
+        self.product_4.uom_id = self.uom_unit
+        self.planning_bom = self.env['mrp.bom'].create({
+            'product_id': self.product_4.id,
+            'product_tmpl_id': self.product_4.product_tmpl_id.id,
+            'product_uom_id': self.uom_unit.id,
+            'product_qty': 4.0,
+            'routing_id': self.routing_1.id,
+            'type': 'normal',
+            'bom_line_ids': [
+                (0, 0, {'product_id': self.product_2.id, 'product_qty': 2}),
+                (0, 0, {'product_id': self.product_1.id, 'product_qty': 4})
+            ]})
```

Comment:
you just added ~10 seconds on runbot, we could maybe avoid that by moving all these master data creation in the setupclass? can you test?